### PR TITLE
Release automation

### PR DESCRIPTION
**Pull request by sammacbeth**
**Reviewer:**

## Description:

Adds two CI jobs to improve the release process:

 1. **Stage Release** - a manually triggered pipeline which will create a PR to `develop` to prepare for a new release. When triggering the release DRI provides the version number to bump to.
Screenshot 2021-08-13 at 14 04 17

 2. **Publish Builds** - triggered when a tag is pushed: creates extension builds for all platforms. This allows the release DRI to generate a release without needing to have a build environment set up.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Publish Builds
+
+on:
+  release:
+    types: [published]
```
Comment (sammacbeth) [inline .github/workflows/publish.yml:5]: Still not sure whether it's better to trigger on tag or release - perhaps a tag is easier to create?

File: scripts/bumpVersion.js [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-const seconds = (time.getUTCHours() * 60 * 60) + (time.getUTCMinutes() * 60) + time.getUTCSeconds();
-const newVersion = `${time.getUTCFullYear()}.${time.getUTCMonth()+1}.${time.getUTCDate()}.${seconds}`
+const newVersion = process.argv[2]
 
-for (let browser of ['chrome', 'firefox']) {
+for (const browser of ['chrome', 'firefox']) {
     const manifest = fs.readFileSync(`browsers/${browser}/manifest.json`, 'utf8')
-    const manifestData = JSON.parse(manifest);
-    manifestData.version = newVersion;
+    const manifestData = JSON.parse(manifest)
```
Comment (jonathanKingston) [inline scripts/bumpVersion.js:7]: Cries in semicolon... sorry :D

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Dependencies
+        run: |
+          npm ci
+          npm i -g web-ext
+      - name: Build
+        run: |
+          npm run generate-etags
```
Comment (jonathanKingston) [inline .github/workflows/publish.yml:32]: I wonder if these three calls should be a script itself? Or perhaps remove the package.json duplication if it's never going to be manual?

File: scripts/bumpVersion.js (hunk 2 above)
Comment (sammacbeth) [reply scripts/bumpVersion.js:7]: linter says no

**Comment by jonathanKingston** [comment]
This is great thank you! I'm going to defer to @GioSensation as he's also done the release process but the code seems fine. I left one comment about maybe moving some stuff to a script instead to reduce duplication.

**Review by jonathanKingston — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -26,8 +26,7 @@
         "release-firefox": "make browser=firefox type=release",
         "dev-chrome": "make dev browser=chrome type=dev watch=1",
         "beta-firefox": "make beta-firefox browser=firefox type=release",
-        "release-chrome": "make browser=chrome type=release && make chrome-release-zip",
-        "release-update": "git submodule update --remote shared/privacy-configuration/ && npm run generate-etags && node scripts/bumpVersion.js"
+        "release-chrome": "make browser=chrome type=release && make chrome-release-zip"
```
Comment (GioSensation) [inline package.json:29]: I don't think this change is intentional. I'll revert it.

cc @jonathanKingston 

```suggestion
        "release-chrome": "make browser=chrome type=release && make chrome-release-zip",
        "release-update": "git submodule update --remote shared/privacy-configuration/ && npm run generate-etags && node scripts/bumpVersion.js"
```

Comment (jonathanKingston) [reply package.json]: I think the idea is to only use this in tagging and shouldn't need to be done manually if that's followed.

Comment (GioSensation) [reply package.json]: Oh, yeah, I see it now. I'll un-revert it 🙃🙂. Thanks!

**Review by GioSensation — APPROVED** [review]
Looks good. Thanks for doing this, Sam!

**Pull request merged**